Packs/SingleConnect/Integrations/SingleConnect/SingleConnect.py

```diff
@@ -0,0 +1,199 @@
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+    """
+
+    try:
+        response = client.list_all_sapm_accounts()
+        search_results = response.get("searchResults")
+        if search_results:
+            if len(search_results) > 0:
+                sapmAccount = search_results[0]
+                if not sapmAccount.get("dbId"):
+                    raise Exception("SAPM Accounts from Single Connect are missing mandatory fields: dbId")
+            else:
+                return "There are no SAPM Accounts but connection is ok"
+            return "ok"
+        else:
+            raise Exception("Unexpected response format")
+    except Exception as e:
+        if 'Forbidden' in str(e) or 'Authorization' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+
+
+def list_all_sapm_accounts_command(client: Client) -> CommandResults:
+    response = client.list_all_sapm_accounts()
+    if response.get("searchResults"):
+        return CommandResults(
+            outputs_prefix='SingleConnect.SapmAccount',
+            outputs_key_field='dbId',
+            outputs=response.get("searchResults")
+        )
+    else:
+        raise Exception("Unexpected response format")
+
+
+def search_sapm_with_secret_name_command(client: Client, secret_name: str) -> CommandResults:
+    response = client.search_sapm_with_secret_name(secret_name)
+    if response.get("searchResults"):
+        return CommandResults(
+            outputs_prefix='SingleConnect.SapmAccount',
+            outputs_key_field='dbId',
+            outputs=response.get("searchResults")
+        )
+    else:
+        raise Exception("Unexpected response format")
+
+
+def show_password_command(client: Client, password_expiration_in_minute=int, sapm_db_id=int,
+                          comment=str) -> CommandResults:
+    response = client.show_password(password_expiration_in_minute, sapm_db_id, comment)
+    return CommandResults(
+        outputs_prefix='SingleConnect.SapmAccount',
+        outputs_key_field='dbId',
+        outputs=response
+    )
+
+
+def get_sapm_user_info_command(client: Client, device_ip: str) -> CommandResults:
+    response = client.get_sapm_user_info(device_ip)
+    return CommandResults(
+        outputs_prefix='SingleConnect.SapmAccount',
+        outputs_key_field='dbId',
+        outputs=response
+    )
+
+
+''' MAIN FUNCTION '''
+
+
+def main(params: dict, args: dict, command: str) -> None:
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+    username = params.get('credentials', {}).get('identifier')
```

Comment:
```suggestion
    params: Dict[str, Any] = demisto.params()
    args: Dict[str, Any] = demisto.args()
    username = params.get('credentials', {}).get('identifier')
```

Reply:
Hi @rshunim, I've added the unit tests and made some overall improvements